src/fides/api/ops/task/graph_task.py

```diff
@@ -765,34 +774,51 @@ def termination_fn(*dependent_values: int) -> Tuple[int, ...]:
                 access_request_data.get(
                     str(k), []
                 ),  # Pass in the results of the access request for this collection
-                *[
+                [
                     access_request_data.get(
                         str(upstream_key), []
                     )  # Additionally pass in the original input data we used for the access request. It's helpful in
                     # cases like the EmailConnector where the access request doesn't actually retrieve data.
                     for upstream_key in t.input_keys
                 ],
+                *_evaluate_erasure_dependencies(t, end_nodes),
             )
             for k, t in env.items()
         }
+
         # terminator function waits for all keys
-        dsk[TERMINATOR_ADDRESS] = (termination_fn, *env.keys())
+        dsk[ROOT_COLLECTION_ADDRESS] = 0
+        dsk[TERMINATOR_ADDRESS] = (termination_fn, *end_nodes)
         update_erasure_mapping_from_cache(dsk, resources, start_function)
         await fideslog_graph_rerun(
             prepare_rerun_graph_analytics_event(
                 privacy_request, env, end_nodes, resources, ActionType.erasure
             )
         )
+
         v = delayed(get(dsk, TERMINATOR_ADDRESS, num_workers=1))
+        update_cts: Dict[str, int] = v.compute()
+        erasure_update_map = {
+            collection.value: update_cts.get(collection.value, 0) for collection in env
+        }
+        return erasure_update_map
 
-        update_cts: Tuple[int, ...] = v.compute()
-        # we combine the output of the termination function with the input keys to provide
-        # a map of {collection_name: records_updated}:
-        erasure_update_map: Dict[str, int] = dict(
-            zip([str(x) for x in env], update_cts)
-        )
 
-        return erasure_update_map
+def _evaluate_erasure_dependencies(
+    t: GraphTask, end_nodes: List[CollectionAddress]
+) -> Set[CollectionAddress]:
+    """
+    Return a set of collection addresses corresponding to collections that need
+    to be erased before the given task. Remove the dependent collection addresses
+    from `end_nodes` so they can be executed in the correct order.
+    """
+    erase_after = t.traversal_node.node.collection.erase_after
+    for collection in erase_after:
+        if collection in end_nodes:
+            end_nodes.remove(collection)
+    # this task will execute after the collections in `erase_after` or
+    # execute at the beginning by linking it to the root node
+    return erase_after if len(erase_after) else set([ROOT_COLLECTION_ADDRESS])
```

Comment:
For this iteration, I'm only considering the `erase_after` field defined at the collection level. I originally had it at the dataset level as well but that would have required modifying `DatasetMetadata` in the fideslang repo